src/background/restrictUnauthenticatedUrlAccess.ts

```diff
@@ -76,26 +114,18 @@ async function initRestrictUnauthenticatedUrlAccess(): Promise<void> {
     reportError(error);
   }
 
-  browser.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
-    if (!changeInfo.url || !tab?.url) {
-      return;
-    }
-
-    const [linked, cachedAuthUrlPatterns] = await Promise.all([
-      isLinked(),
-      sessionValue.get(),
-    ]);
-
-    if (linked || !cachedAuthUrlPatterns) {
-      return;
-    }
-
-    const isRestrictedUrl = testMatchPatterns(cachedAuthUrlPatterns, tab.url);
+  if (!(await isLinked())) {
+    browser.tabs.onUpdated.addListener(handleRestrictedTab);
+  }
 
-    if (isRestrictedUrl) {
-      await browser.tabs.update(tabId, {
-        url: ssoUrl ?? getDefaultAuthUrl(tab.url),
-      });
+  addAuthListener(async (auth) => {
+    if (auth) {
+      browser.tabs.onUpdated.removeListener(handleRestrictedTab);
+    } else {
+      await forEachTab(async ({ tabId, url }) =>
+        redirectRestrictedTab({ tabId, url }),
+      );
```

Comment:
Nit

Direct references as callbacks are only a problem on Array methods, where the second parameter is a generic `number`: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md

Since we control `forEachTab`, we have no reason to add parameters instead of just adding properties to the first object. So I've been passing references directly

```suggestion
      await forEachTab(redirectRestrictedTab);
```

Reply:
Thanks for catching this - I think this was an artifact of a refactor I forgot to accomodate